### PR TITLE
Function param types

### LINKED ISSUE #7589: Method parameters that are functions should explicitly declare their signiture
* Electron version: N/A
* Operating system: N/A

All parameters that are functions should declare the parameters that they will provide when called.  We should follow the same syntax as the inline Object properties declaration.  Some methods already do this, such as:

https://github.com/electron/electron/blob/master/docs/api/app.md#appimportcertificateoptions-callback-linux

However others do not: 

https://github.com/electron/electron/blob/master/docs/api/app.md#appmakesingleinstancecallback

Once standardized we can reuse the object properties parser in `electron-docs-linter` to parse the function parameters.

/cc @zeke

### PR DESCRIPTION
**Pull request by MarshallOfSound**
Fixes #7589

/cc @zeke

### CODE REVIEW

File: docs/api/structures/upload-data.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# ThumbarButton Object
```
Comment (zeke) [inline docs/api/structures/upload-data.md]: should be "UploadData", right?

Comment (zeke) [inline docs/api/structures/upload-data.md]: or "Upload Data"?

Comment (MarshallOfSound) [inline docs/api/structures/upload-data.md]: Fixed 👍

File: docs/api/structures/desktop-capturer-source.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Desktop Capturer Source Object
```
Comment (zeke) [inline docs/api/structures/desktop-capturer-source.md]: Should this be "DesktopCapturerSource Object"?

I don't feel strongly one way or the other, but I think so far we've mostly been CamelCasing them (or whatever all-leading-caps case is called 🤔 ).

Comment (MarshallOfSound) [inline docs/api/structures/desktop-capturer-source.md]: Fixed 👍

File: docs/api/desktop-capturer.md [hunk 3]
```diff
@@ -61,25 +61,14 @@ The `desktopCapturer` module has the following methods:
...
-* `name` String - A screen source will be named either `Entire Screen` or
-  `Screen <index>`, while the name of a window source will match the window
-  title.
-* `thumbnail` [NativeImage](native-image.md) - A thumbnail image. **Note:**
-  There is no guarantee that the size of the thumbnail is the same as the
-  `thumnbailSize` specified in the `options` passed to
-  `desktopCapturer.getSources`. The actual size depends on the scale of the
-  screen or window.
+`sources` is an array of [`DesktopCapturerSource`](structures/desktop-capturer-source.md)
+objects, each `Source` represents a screen or an individual window that can be
```
Comment (kevinsawicki) [inline docs/api/desktop-capturer.md]: > each `Source` represents...

Should this be `DesktopCapturerSource` instead?

Comment (MarshallOfSound) [inline docs/api/desktop-capturer.md]: 👍

**Comment by kevinsawicki** [comment]
Looks good to me, thanks for adding these 👍 👍 👍

**Pull request merged**